client/src/elm/Pages/WellChild/Activity/Utils.elm

```diff
@@ -389,23 +389,26 @@ expectNutritionAssessmentTask currentDate assembled task =
             True
 
 
-mandatoryNutritionAssessmentTasksCompleted : NominalDate -> AssembledData -> Bool
-mandatoryNutritionAssessmentTasksCompleted currentDate assembled =
-    resolveMandatoryNutritionAssessmentTasks currentDate assembled
-        |> List.filter (not << nutritionAssessmentTaskCompleted currentDate assembled)
-        |> List.isEmpty
+mandatoryNutritionAssessmentTasksCompleted : NominalDate -> Site -> AssembledData -> Bool
+mandatoryNutritionAssessmentTasksCompleted currentDate site assembled =
+    resolveMandatoryNutritionAssessmentTasks currentDate site assembled
+        |> List.all (nutritionAssessmentTaskCompleted currentDate assembled)
 
 
-resolveMandatoryNutritionAssessmentTasks : NominalDate -> AssembledData -> List NutritionAssessmentTask
-resolveMandatoryNutritionAssessmentTasks currentDate assembled =
+resolveMandatoryNutritionAssessmentTasks : NominalDate -> Site -> AssembledData -> List NutritionAssessmentTask
+resolveMandatoryNutritionAssessmentTasks currentDate site assembled =
     List.filter (expectNutritionAssessmentTask currentDate assembled) <|
         case assembled.encounter.encounterType of
             PediatricCare ->
                 [ TaskHeight, TaskHeadCircumference, TaskMuac, TaskNutrition, TaskWeight ]
 
             _ ->
-                -- Height is optional for CHW.
-                [ TaskHeadCircumference, TaskMuac, TaskNutrition, TaskWeight ]
+                if site == SiteBurundi then
+                    -- Height is optional for CHW.
```

Comment:
This comment is not only for Burundi. Move above if.
Add specific comment for Burundi about weight.